src/pages/workspace/members/WorkspaceOwnerChangeCheck.tsx

```diff
@@ -33,9 +33,33 @@ type WorkspaceOwnerChangeCheckProps = WorkspaceOwnerChangeCheckOnyxProps & {
 function WorkspaceOwnerChangeCheck({personalDetails, policy, accountID, error}: WorkspaceOwnerChangeCheckProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const [displayTexts, setDisplayTexts] = useState({
+        title: '',
+        text: '',
+        buttonText: '',
+    });
 
     const policyID = policy?.id ?? '';
 
+    const updateDisplayTexts = useCallback(() => {
+        const changeOwnerErrors = Object.keys(policy?.errorFields?.changeOwner ?? {});
+        if (error !== changeOwnerErrors[0]) {
+            return;
+        }
+
+        const texts = WorkspaceSettingsUtils.getOwnershipChecksDisplayText(error, translate, policy, personalDetails?.[accountID]?.login);
+        setDisplayTexts(texts);
+    }, [accountID, error, personalDetails, policy, translate]);
+
+    useEffect(() => {
+        updateDisplayTexts();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        updateDisplayTexts();
+    }, [updateDisplayTexts]);
```

Comment:
I can't think of a reason to have both, it seems like we can get rid of the first one without affecting functionality.

Reply:
@carlosmiceli I started investigating components that disable `react-hooks/exhaustive-deps` and `react-compiler/react-compiler` because of [this](https://github.com/Expensify/App/pull/51480#discussion_r1824439171) comment. That's how I noticed this weird duplicate.

[I created a follow up issue](https://github.com/Expensify/App/issues/51972) to react-compiler upgrade [PR](https://github.com/Expensify/App/pull/51183), could you assign me to it? I don't want to spend too much time on this, but to fix some of obvious mistakes we have in the codebase. I could fix this one there too, let me know what do you think :)

Reply:
Not sure if you meant to tag @carlosmiceli, but I assigned you to the issue above.

Reply:
Ahh, sorry. I meant to tag you of cource 🤦